src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+	 *
+	 *     There are entries in the bar that are never modified by changing context,
+	 *     but may vary according to e.g. the current user's permissions or settings
+	 *     of the installation. There also is the tools-section of entries in the bar
+	 *     that is used to show tools that are opened on request of the user, e.g. the
+	 *     help, or depending on requirements of the content, e.g. a local navigation.
+	 *
+	 *   composition: >
+	 *     The Mainbar holds Slates and Bulky Buttons.
+	 *
+	 *     In a desktop environment, a vertical bar is rendered on the left side
+	 *     of the screen covering the full height (minus header- and footer area).
+	 *     Entries are aligned vertically.
+	 *
+	 *     In a mobile context, the bar will be rendered horizontally on the bottom.
+	 *
+	 *     When the entries of a Mainbar exceed the available height (mobile: width),
+	 *     remaining buttons will be collected in a "..."-Button.
+	 *
+	 *     The Mainbar is always visible and available (except in specialized views
+	 *     like the exam mode) as a static screen element unaffected by scrolling.
+     *
+	 *   effect: >
+     *     Clicking an entry will carry out its configured action. For slates, this
+	 *     is expanding the slate, while for links an according page opens.
+	 *
+	 *     Buttons in the Mainbar are stateful, i.e. they have a pressed-status
+	 *     that can either be toggled by clicking the same button again or by
+	 *     clicking a different button. This does not apply to Buttons that direcly
+	 *     change the page.
+	 *
+	 *     Opening a slate by clicking an entry will close all other slates in the
+	 *     Mainbar. On desktop, slates open on the right hand of the Mainbar, between
+	 *     bar and content, thus "pushing" the content to the right, if there is not
+	 *     enough room.
+	 *
+	 *     If the content's width would fall below its defined minimum, the expanded
+	 *     slate is opened above (like in overlay, not "on top of") the content.
+	 *
+	 *     The slates height equals that of the Mainbar. Also, their position will
+	 *     remain static when the page is scrolled. A button to close a slate is
+	 *     rendered underneath the slate. It will  close all visible Slates and reset
+	 *     the states of all mainbar-entries.
+	 *
+	 *     When a tool (such as the help), whose contents are displayed in a slate,
+	 *     is being triggered, a special entry is rendered as first element of the
+	 *     Mainbar, making the available/invoked tool(s) accessible. Tools can be
+	 *     closed, i.e. removed from the Mainbar, via a Close Button. When the last
+	 *     Tool is closed, the tools-section is removed as well.
+	 *
+	 *   rivals:
+	 *     Tab Bar: >
+	 *       The Mainbar (and its components) shall not be used to substitute
+	 *       functionality available at objects, such as settings, members or
+	 *       learning progress. Those remain in the Tab Bar.
```

Comment:
So many bars ;-) This one is not existing in the KS yet. Personally I would prefer to not add rules to non existing KS components.

Reply:
I see the point of @alex40724 especially the "Content Actions" and "Personal Desktop" points should just move to purpose. I am not even sure if things with such a scope will ever be KS Items. The Tab Bar is less problematic and IMO might stay here for the moment. However, we have to make ourselves a mental note to rename/adapt them as soon as we have the Tab Bar.